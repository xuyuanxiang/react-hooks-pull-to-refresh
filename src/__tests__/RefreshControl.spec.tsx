import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import 'jest-styled-components';
import { RefreshControl, RefreshControlProvider, RefreshState } from '../';

describe('<RefreshControl/>', () => {
  let root = document.createElement('div');

  beforeEach(() => {
    document.body.appendChild(root);
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(root);
    document.body.removeChild(root);
  });

  it('should render default props', () => {
    act(() => {
      ReactDOM.render(
        <RefreshControlProvider value={{ state: RefreshState.DID_MOUNT }}>
          <RefreshControl />
        </RefreshControlProvider>,
        root,
      );
    });
    expect(document.documentElement).toMatchSnapshot();
  });
});
