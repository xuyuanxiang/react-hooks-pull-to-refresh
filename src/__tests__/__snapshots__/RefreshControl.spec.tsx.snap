// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RefreshControl/> should render default props 1`] = `
.c0 {
  background-color: #f6f6f6;
  text-align: center;
  padding: 32px;
}

<html>
  <head>
    <style
      data-styled="active"
      data-styled-version="5.0.1"
    >
      .c0{background-color:#f6f6f6;text-align:center;padding:32px;}
    </style>
  </head>
  <body>
    <div>
      <div
        class="c0"
      >
        <p>
          下拉可以刷新
        </p>
      </div>
    </div>
  </body>
</html>
`;
